node/src/components/consensus/highway_core/state.rs

```diff
@@ -141,7 +142,7 @@ impl<C: Context> Fault<C> {
 /// Both observers and active validators must instantiate this, pass in all incoming vertices from
 /// peers, and use a [FinalityDetector](../finality_detector/struct.FinalityDetector.html) to
 /// determine the outcome of the consensus process.
-#[derive(Debug, Clone, DataSize)]
+#[derive(Debug, Deserialize, Clone, DataSize, Serialize)]
```

Comment:
Why do we need `Deserialize` for these types? Do we make use of these automatically derived deserializers in the client that reads the dumped state of consensus?

Reply:
That was the initial intention, but I have copied these types in the interim. I'll remove the `Deserialize`.